v03_pipeline/lib/reference_data/config.py

```diff
@@ -211,6 +224,12 @@ def custom_mpc_select(ht):
             'version': '2.9.3',
             'path': 'gs://seqr-reference-data/GRCh37/dbNSFP/v2.9.3/dbNSFP2.9.3_variant.with_new_scores.ht',
             'custom_select': dbnsfp_custom_select,
+            'custom_select_keys': [
```

Comment:
I added `custom_select_keys` because I needed a way to get these keys for validation and couldn't get them from the `custom_select` functions.

Reply:
Yep!  Love the idea.  Would this capture a change to the logic in a custom select though? Say we change:

```
selects['MutPred_score'] = hl.parse_float32(ht.MutPred_score)
```

to 

```
selects['MutPred_score'] = hl.int32(hl.parse_float32(ht.MutPred_score))
```
or something.

Reply:
Since we're checking the config selects the joined table selects, we only need to check if the fields from or the left hand of that equals expression from your comment are in the joined table. 

Do we also need to compare the fields on the right hand side of the expression with the fields on the dataset tables?

Reply:
it does make sense to couple this list of keys with the `custom_select` functions though so if they ever change they only change in one spot

Reply:
I think we probably should do something to ensure the logic of the function itself has changed or not changed.  My initial thought would be to do something like hash the definition of the function itself so, so if we change the source code in any way we'd be "out of sync".  It might be over-engineering (or even under-engineering) though.

Something like this works:
```
def f():
	pass

import inspect
print(inspect.getsource(f))
```

Reply:
if we went this route, would we store the hashed function definition in the config?

Reply:
this is the only pending piece for me on this issue

Reply:
I think we'd need to store it in the globals actually.  It actually might be worth chatting about this offline

Reply:
We chatted offline and decided only to check the resulting selected fields and _not_ the expressions that select the fields from the dataset_ht.